### PR TITLE
use disallow-any-generics mypy flag

### PR DESCRIPTION
**Pull request by vtomole**
Part of https://github.com/SupertechLabs/general-superstaq/issues/68

### CODE REVIEW

**Review by SalahedeenIssa — APPROVED** [review]
LGTM

File: cirq_superstaq/custom_gates.py [hunk 1]
```diff
@@ -31,7 +33,7 @@ class ZZSwapGate(cirq.Gate, cirq.ops.gate_features.InterchangeableQubitsGate):
     For :math:`\theta = 0`, the ZZ-SWAP gate is just an ordinary SWAP.
     """
 
-    def __init__(self, theta: cirq.TParamVal) -> None:
+    def __init__(self, theta: Optional[float] = None) -> None:
```
Comment (richrines1) [inline cirq_superstaq/custom_gates.py:36]: why this change? I made this gate work with sympy expressions

File: cirq_superstaq/custom_gates.py [hunk 2]
```diff
@@ -445,7 +447,7 @@ def phi(self) -> float:
     def theta(self) -> float:
         return self.exponent * np.pi
 
-    def __pow__(self, power: float) -> "RGate":
+    def __pow__(self, power: Union[float, sympy.Expr]) -> "RGate":
```
Comment (richrines1) [inline cirq_superstaq/custom_gates.py:450]: fine as is, but fwiw this is equivalent to
```suggestion
    def __pow__(self, power: cirq.TParamVal) -> "RGate":
```
(elsewhere as well)

Comment (vtomole) [reply cirq_superstaq/custom_gates.py]: Done.

File: cirq_superstaq/custom_gates.py [hunk 3]
```diff
@@ -445,7 +446,7 @@ def phi(self) -> float:
     def theta(self) -> float:
         return self.exponent * np.pi
 
-    def __pow__(self, power: float) -> "RGate":
+    def __pow__(self, power: cirq.TParamVal) -> "RGate":
```
Comment (richrines1) [inline cirq_superstaq/custom_gates.py:449]: i think this one should be reverted - `RGate.__init__` only accepts floats (for now)
```suggestion
    def __pow__(self, power: float) -> "RGate":
```

File: cirq_superstaq/job_test.py [hunk 4]
```diff
@@ -31,8 +32,13 @@ def new_job() -> css.Job:
     return css.Job(client, "my_id")
 
 
-def mocked_get_job_requests(*job_dicts: Dict[str, Any]) -> mock._patch:
-    """Mocks the server's response to `get_job` requests using the given sequence of job_dicts."""
+def mocked_get_job_requests(
+    *job_dicts: Dict[str, Any]
+) -> "mock._patch[Union[MagicMock, AsyncMock]]":
```
Comment (richrines1) [inline cirq_superstaq/job_test.py:37]: nit
```suggestion
) -> "mock._patch[Union[mock.MagicMock, mock.AsyncMock]]":
```
(we use "mock.MagicMock" elsewhere in this file so it feels weird to import it separately for this line)

Comment (richrines1) [reply cirq_superstaq/job_test.py:37]: also fwiw - not sure which is more correct in this case, but it seems to pass type check for me with either
```suggestion
) -> "mock._patch[mock.MagicMock]":
```
or
```suggestion
) -> "mock._patch[mock.Mock]":
```

Comment (vtomole) [reply cirq_superstaq/job_test.py:37]: Will do the last suggestion.

File: cirq_superstaq/compiler_output.py [hunk 5]
```diff
@@ -32,7 +32,7 @@ def __init__(
         pulse_sequences: Optional[Any] = None,
         seq: Optional["qtrl.sequencer.Sequence"] = None,
         jaqal_programs: Optional[Union[List[str], str]] = None,
-        pulse_lists: Optional[Union[List[List], List[List[List]]]] = None,
+        pulse_lists: Optional[Union[List[List[str]], List[List[List[str]]]]] = None,
```
Comment (richrines1) [inline cirq_superstaq/compiler_output.py:35]:
```suggestion
        pulse_lists: Optional[Union[List[List[List[Any]]], List[List[List[List[Any]]]]]] = None,
```

https://github.com/SupertechLabs/qiskit-superstaq/pull/234#discussion_r1006241901

File: cirq_superstaq/custom_gates.py (hunk 3 above)
Comment (richrines1) [reply cirq_superstaq/custom_gates.py:466]: should this have been resolved? it seems like either this signature or `RGate.__init__`'s is wrong

**Review by SalahedeenIssa — DISMISSED** [review]

File: cirq_superstaq/custom_gates.py (hunk 3 above)
Comment (vtomole) [reply cirq_superstaq/custom_gates.py]: Applied your suggestion. Mypy still passes.

**Review by richrines1 — APPROVED** [review]
lgtm!
